web/src/main/java/org/akaza/openclinica/control/admin/DownloadVersionSpreadSheetServlet.java

```diff
@@ -90,16 +91,18 @@ public void processRequest() throws Exception {
             // IOUtils.copy(getCoreResources().getInputStream(CRF_VERSION_TEMPLATE), fos);
             // IOUtils.closeQuietly(fos);
         } else {
-            excelFile = new File(dir + excelFileName);
+        	// fix path traversal issue
+            excelFile = new File(dir,excelFileName);
             // backwards compat
-            File oldExcelFile = new File(dir + oldExcelFileName);
-            if (oldExcelFile.exists() && oldExcelFile.length() > 0) {
-                if (!excelFile.exists() || excelFile.length() <= 0) {
-                    // if the old name exists and the new name does not...
-                    excelFile = oldExcelFile;
-                    excelFileName = oldExcelFileName;
-                }
-            }
+            File oldExcelFile = new File(dir, oldExcelFileName);            
```

Comment:
before continuing, I would check for canonical path (see below) and only allow if the canonical path starts with the base dir path.

Reply:
changed--check for canonical path  and only allow if the canonical path starts with the base dir path.